src/backend/marsha/core/tests/test_api_video_share_live_media.py

```diff
@@ -0,0 +1,1190 @@
+            mock_broadcast_message.assert_called_with(
+                str(video.id), json.dumps(content)
+            )
+
+    @override_settings(LIVE_CHAT_ENABLED=True)
+    @override_settings(XMPP_BOSH_URL="https://xmpp-server.com/http-bind")
+    @override_settings(XMPP_CONFERENCE_DOMAIN="conference.xmpp-server.com")
+    @override_settings(XMPP_DOMAIN="conference.xmpp-server.com")
+    @override_settings(XMPP_JWT_SHARED_SECRET="xmpp_shared_secret")
+    def test_api_video_shared_live_media_start_not_ready(self):
+        """An instructor can not start a not ready shared live media."""
+
+        video = VideoFactory(
+            playlist__title="foo bar",
+            playlist__lti_id="course-v1:ufr+mathematics+00001",
+            upload_state=PENDING,
+            live_state=RUNNING,
+            live_type=JITSI,
+            live_info={
+                "medialive": {
+                    "input": {
+                        "id": "medialive_input_1",
+                        "endpoints": [
+                            "https://live_endpoint1",
+                            "https://live_endpoint2",
+                        ],
+                    },
+                    "channel": {"id": "medialive_channel_1"},
+                },
+                "mediapackage": {
+                    "id": "mediapackage_channel_1",
+                    "endpoints": {
+                        "hls": {
+                            "id": "endpoint1",
+                            "url": "https://channel_endpoint1/live.m3u8",
+                        },
+                    },
+                },
+            },
+        )
+
+        for state in [
+            PENDING,
+            PROCESSING,
+            HARVESTING,
+            ERROR,
+            HARVESTED,
+            DELETED,
+        ]:
+            shared_live_media = SharedLiveMediaFactory(
+                extension="pdf",
+                title="slides",
+                # upload_state=random.choice([s[0] for s in STATE_CHOICES if s[0] != READY]),
+                upload_state=state,
+                uploaded_on=datetime(2021, 11, 30, tzinfo=pytz.utc),
+                nb_pages=3,
+            )
+            video.shared_live_medias.add(shared_live_media)
+
+            jwt_token = AccessToken()
+            jwt_token.payload["resource_id"] = str(shared_live_media.video.id)
+            jwt_token.payload["roles"] = [
+                random.choice(["instructor", "administrator"])
+            ]
+            jwt_token.payload["permissions"] = {"can_update": True}
+            jwt_token.payload["user"] = {"id": "56255f3807599c377bf0e5bf072359fd"}
+
+            with mock.patch.object(
+                api.video, "broadcast_message"
+            ) as mock_broadcast_message, mock.patch(
+                "marsha.core.serializers.xmpp_utils.generate_jwt"
+            ) as mock_jwt_encode:
+                mock_jwt_encode.return_value = "xmpp_jwt"
```

Comment:
```suggestion
```

Don't patch them it's not needed. Your test will fail if these functions are reached.

Reply:
I keep mock_broadcast_message as I'm testing it has not been called.